rmc/utils/constraint.py

```diff
@@ -1408,39 +1455,47 @@ def create_context_with_oe(
         - `context_with_oe_dedup`: Deduplicated version of `context_with_oe` that only contains missense o/e and transcript annotations.
 
     :param freeze: RMC data freeze number.
-    :param str missense_str: String representing missense variant consequence. Default is MISSENSE.
-    :param int n_partitions: Number of desired partitions for the VEP context Table.
+    :param filter_to_canonical: Whether to filter to canonical transcripts only. Default is False.
+    :param missense_str: String representing missense variant consequence. Default is MISSENSE.
+    :param n_partitions: Number of desired partitions for the VEP context Table.
         Repartition VEP context Table to this number on read.
         Default is 30000.
-    :param bool overwrite_temp: Whether to overwrite intermediate temporary (OE-independent) data if it already exists.
+    :param overwrite_temp: Whether to overwrite intermediate temporary (OE-independent) data if it already exists.
         If False, will read existing intermediate temporary data rather than overwriting.
         Default is False.
+    :param filter_outliers: Whether to filter out outlier transcripts. Default is False.
+    :param vep_version: VEP version to use. Default is "105".
     :return: None; function writes Table to resource path.
     """
-    logger.info("Importing set of transcripts to keep...")
-    transcripts = get_constraint_transcripts(outlier=False)
-
     logger.info(
         "Reading in SNPs-only, VEP-annotated context HT and filtering to missense"
         " variants in canonical transcripts..."
     )
     # Using vep_context.path to read in table with fewer partitions
     # VEP context resource has 62164 partitions
     ht = (
-        hl.read_table(vep_context.path, _n_partitions=n_partitions)
+        hl.read_table(vep_context.versions[vep_version].path)
         .select_globals()
         .select("vep", "was_split")
     )
+    ht = ht.naive_coalesce(n_partitions)
     ht = filter_vep_transcript_csqs(
         t=ht,
         vep_root="vep",
         synonymous=False,
-        canonical=True,
+        canonical=filter_to_canonical,
+        protein_coding=True,
         ensembl_only=True,
         filter_empty_csq=True,
         csqs={missense_str},
     )
-    ht = ht.filter(transcripts.contains(ht.transcript_consequences.transcript_id))
+    if filter_outliers:
+        logger.info("Importing set of transcripts to keep...")
+        transcripts = get_constraint_transcripts(outlier=False)
```

Comment:
I checked our RMC results rather than going back to the context HT to make this check quicker -- 
```
get_constraint_transcripts(all_transcripts=True, filter_to_canonical=True)
```
returns 18623 transcripts,
```
gs://regional_missense_constraint/constraint/4.1/1/rmc_browser.ht
```
has 7448 transcripts, and
```
gs://regional_missense_constraint/constraint/4.1/1/no_breaks.he
```
has 11973 transcripts.

7410 / 7448 of the RMC transcripts and 11212 / 11973 of the no break transcripts overlap the 18623 transcripts. Spot checking a few of the transcripts unique to the RMC results / no breaks shows that they don't have constraint metrics in the browser and likely aren't protein coding transcripts (so an artifact of this first test run, where I forgot to filter to protein-coding transcripts).

There is 1 [transcript](https://gnomad.broadinstitute.org/transcript/ENST00000634965?dataset=gnomad_r4) in the 18623 transcripts that is unaccounted for in our results (not in RMC results or no breaks results), but it doesn't have a defined gene symbol, so I'm not 100% sure why it's included in the gene constraint results table

Reply:
Ok cool, so in theory now that we have switched from filtering on ENST to protein_coding, this should be resolved, minus that 1 errant transcript? So it should be 18622 transcripts total